src/dialogs/more-info/ha-more-info-tab-history.ts

```diff
@@ -0,0 +1,187 @@
+import "../../panels/logbook/ha-logbook";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+
+@customElement("ha-more-info-tab-history")
+export class MoreInfoTabHistoryDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public entityId!: string;
+
+  @internalProperty() private _stateHistory?: HistoryResult;
+
+  @internalProperty() private _isLoading = false;
+
+  private _entries: LogbookEntry[] = [];
+
+  private _userIdToName = {};
+
+  private _historyRefreshInterval?: number;
+
+  protected render(): TemplateResult {
+    if (!this.entityId) {
+      return html``;
+    }
+    const entityId = this.entityId;
+    const stateObj = this.hass.states[entityId];
+
+    if (!stateObj) {
+      return html``;
+    }
+
+    return html`
+      ${this._isLoading
+        ? html`
+            <ha-circular-progress
+              active
+              alt=${this.hass.localize("ui.common.loading")}
+            ></ha-circular-progress>
+          `
+        : html`
+            <state-history-charts
+              up-to-now
+              .hass=${this.hass}
+              .historyData=${this._stateHistory}
+            ></state-history-charts>
+            <ha-logbook
+              narrow
+              .hass=${this.hass}
+              .entries=${this._entries}
+              .userIdToName=${this._userIdToName}
+            ></ha-logbook>
+          `}
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    if (!this.entityId) {
+      clearInterval(this._historyRefreshInterval);
+    }
+
+    if (changedProps.has("entityId")) {
+      this._refreshData();
+    }
+  }
+
+  private async _refreshData(): Promise<void> {
+    this._isLoading = true;
+    const fetchUserDone = this._fetchUserNames();
+    const fetchHistoryDone = this._getStateHistory();
+    const fetchLogBookDone = this._getLogBookData();
+
+    await Promise.all([fetchHistoryDone, fetchLogBookDone, fetchUserDone]);
+
+    this._isLoading = false;
+
+    clearInterval(this._historyRefreshInterval);
+    this._historyRefreshInterval = window.setInterval(() => {
+      this._getStateHistory();
+    }, 60 * 1000);
+  }
+
+  private async _getStateHistory(): Promise<void> {
+    this._stateHistory = await getRecentWithCache(
+      this.hass!,
+      this.entityId,
+      {
+        refresh: 60,
+        cacheKey: `more_info.${this.entityId}`,
+        hoursToShow: 24,
+      },
+      this.hass!.localize,
+      this.hass!.language
+    );
+  }
+
+  private async _getLogBookData() {
+    const yesterday = new Date(new Date().getTime() - 24 * 60 * 60 * 1000);
+    const now = new Date();
+    const entries = await getLogbookData(
+      this.hass,
+      yesterday.toISOString(),
+      now.toISOString(),
+      this.entityId,
+      true
+    );
+
+    // @ts-ignore
+    this._entries = entries;
+  }
+
+  private async _fetchUserNames() {
```

Comment:
We got to think of something for this, not funny to load the same data on every more info window.

Reply:
Yea. Do you have an idea?